executor/newdeploy/newdeploy.go

```diff
@@ -58,6 +58,15 @@ func (deploy *NewDeploy) createOrGetDeployment(fn *crd.Function, env *crd.Enviro
 
 	existingDepl, err := deploy.kubernetesClient.ExtensionsV1beta1().Deployments(deployNamespace).Get(deployName, metav1.GetOptions{})
 	if err == nil {
+		// If minScale is 0 then scale to 1, otherwise scale up to minScale
```

Comment:
I am not sure what the API works right now. But given that the kubernetes API is out of our control, it might be good  to be a bit more defensive here, and handle the case `existingDepl == nil && err == nil`. Currently that would result in a NPE